srcjs/input_rate.js

```diff
@@ -251,6 +251,20 @@ var InputDeferDecorator = function(target) {
   };
 }).call(InputDeferDecorator.prototype);
 
+var InputEventDecorator = function(target) {
+  this.target = target;
+};
+(function() {
+  this.setInput = function(name, value, immediate) {
+    var evt = jQuery.Event("shiny:inputchanged");
+    evt.name = name;
```

Comment:
@jcheng5 The `name` here may have a suffix, e.g. `foo:shiny.action` for an action button with id `foo`. Do we want to do anything about it? The real question is, do we want to expose this kind of internal special id's to users?